### PR TITLE
Rspec

### PR DESCRIPTION
**Pull request by geraldgsh**
- Created Rspec test cases for 6 methods as specified in the Odin project

### CODE REVIEW

File: spec/main_spec.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+  describe '#my_each_with_index' do
+    context 'If block is not given' do
+      it 'Return enumerable object' do
+        expect(array1.my_each_with_index.is_a?(Enumerable)).to eql(true)
+      end
+    end
+    context 'If block is given' do
+      it 'Return each element with index in an array' do
+        expect(array2.my_each_with_index { |item, i| "#{item}: #{i}" }).to eql(0..1)
```
Comment (maelfosso) [inline spec/main_spec.rb]: Do not match

```ruby
irb(main):008:0> array2.each_with_index { |item, i| "#{item} : #{i}" }
=> ["star", "wars"]
irb(main):009:0> array2.my_each_with_index { |item, i| "#{item} : #{i}" }
=> 0..1
irb(main):010:0> 
irb(main):025:0> array2.each_with_index { |item, i| "#{item} : #{i}" } == array2.my_each_with_index { |item, i| "#{item} : #{i}" } 
=> false

```

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Not approved : Only one bug to fix :disappointed: 

Hello @geraldgsh @webmarkyn 

Nice work :clap: You really master **RSpec**
There is only one bug to fix. Please, check it.

Kindly make the changes and submit a new code review request.

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com)

**Comment by geraldgsh** [comment]
@maelfosso Noted with thanks. We'll make the amendments according and submit another request for code review.

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great work implementing the suggested changes :clap: . However, you still have one issue to fix in order to get approval. check :arrow_down: 

- I cannot see any **linter checks**, follow this [guide](https://github.com/microverseinc/linters-config/tree/master/ruby)

That's it :relaxed: . When finished, kindly resubmit another code review request :pray: 

**Ahmed Mahfoudh**

**Comment by geraldgsh** [comment]
image

**Comment by geraldgsh** [comment]
@stratospherique Unfortunately stickler for this Repo isn't working despite many attempts to troubleshoot it.

However, Rubocop lint test passed as illustrated in the screenshot above.

**Review by wathigo — APPROVED** [review]
## Hi Devs  :wave:  

**This is approved. You can move on :next_track_button:**  

_Good luck!
Reviewed by: **Simon Wathigo**_

**Comment by geraldgsh** [comment]
Thanks, @wathigo. We'll proceed with the merge.

**Pull request merged**